src/android/VideoEnabledWebChromeClient.java

```diff
@@ -0,0 +1,294 @@
+package org.apache.cordova.inappbrowser;
+
+import android.media.MediaPlayer;
+import android.view.SurfaceView;
+import android.view.View;
+import android.view.ViewGroup;
+import android.webkit.WebChromeClient;
+import android.widget.FrameLayout;
+
+/**
+ * This class serves as a WebChromeClient to be set to a WebView, allowing it to play video.
+ * Video will play differently depending on target API level (in-line, fullscreen, or both).
+ *
+ * It has been tested with the following video classes:
+ * - android.widget.VideoView (typically API level <11)
+ * - android.webkit.HTML5VideoFullScreen$VideoSurfaceView/VideoTextureView (typically API level 11-18)
+ * - com.android.org.chromium.content.browser.ContentVideoView$VideoSurfaceView (typically API level 19+)
+ *
+ * Important notes:
+ * - For API level 11+, android:hardwareAccelerated="true" must be set in the application manifest.
+ * - The invoking activity must call VideoEnabledWebChromeClient's onBackPressed() inside of its own onBackPressed().
+ * - Tested in Android API levels 8-19. Only tested on http://m.youtube.com.
+ *
+ * @author Cristian Perez (http://cpr.name)
```

Comment:
Are you Christian Perez? If not, do you have permission to submit this code under the Apache license?

If yes, then the Apache license needs to be added to this file.

Reply:
This seems copy/pasted from https://github.com/cprcrack/VideoEnabledWebView

Reply:
> This seems copy/pasted from https://github.com/cprcrack/VideoEnabledWebView

It actually is. I needed those changes in my current project so I forked the latest repo and modified it so, if anyone else wants to use it. They can.